.circleci/config.yml

```diff
@@ -1,56 +1,95 @@
+    keys:
+      - v2-git-{{ .Branch }}-{{ .Revision }}
+      - v2-git-{{ .Branch }}
+      - v2-git
+  save-git-cache: &save-git-cache
+    name: Save git cache
+    key: v2-git-{{ .Branch }}-{{ .Revision }}
+    paths:
+      - ".git"
+
+  restore-npm-cache: &restore-npm-cache
+    name: "Restore npm cache"
+    keys:
+       - v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
+       - v1-npmcache-{{ checksum ".nvmrc" }}
+  save-npm-cache: &save-npm-cache
+    name: "Save npm cache"
+    key: v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
+    paths:
+      - ~/.npm
+
+  #
+  # Artifact prep
+  #
+  setup-results-and-artifacts: &setup-results-and-artifacts
+    name: Create Directories for Results and Artifacts
+    command: mkdir -p                  \
+      $CIRCLE_ARTIFACTS                \
+      $CIRCLE_TEST_REPORTS/client
+
+
 version: 2
 jobs:
-  build:
-    parallelism: 6
+  #
+  # Shared setup required for all jobs
+  #
+  setup:
+    docker: *docker-image
     working_directory: ~/wp-calypso
-    docker:
-      - image: circleci/node:10.10.0-browsers
-        environment:
-          CIRCLE_ARTIFACTS: /tmp/artifacts
-          CIRCLE_TEST_REPORTS: /tmp/test_results
-
     steps:
-      - restore_cache:
-          name: Restore git cache
-          key: v1-20180524-git
-
+      - restore_cache: *restore-git-cache
       - checkout
-
-    # Update origin/master so we can diff against it later
+      # Update origin/master so we can diff against it later
       - run:
           name: Update master branch
           command: git fetch --force origin master
+      - save_cache: *save-git-cache
 
-      - save_cache:
-          name: Save git cache
-          key: v1-20180524-git
-          paths:
-            - ".git"
-
-      - run:
-          name: Create Directories for Results and Artifacts
-          command: |
-            mkdir -p $CIRCLE_ARTIFACTS
-            mkdir -p $CIRCLE_TEST_REPORTS
-
-      - restore_cache:
-          name: "Restoring npm cache"
-          keys:
-             - v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
-             - v1-npmcache-{{ checksum ".nvmrc" }}
-
+      - restore_cache: *restore-npm-cache
       - run: npm ci
+      - save_cache: *save-npm-cache
+      - persist_to_workspace:
+          root: .
+          paths: .
+
+  #
+  # Build translations
+  #
+  translate:
+    docker: *docker-image
+    working_directory: ~/wp-calypso
+    environment: *shared-environment
+
+    steps:
+      - attach_workspace:
+          at: .
 
-      - save_cache:
-          name: "Saving npm cache"
-          key: v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
-          paths:
-            - ~/.npm
+      - run: *setup-results-and-artifacts
 
       - run:
           name: Build Server
           environment:
-              NODE_ENV: test
+            NODE_ENV: test
           command: npm run build-server
```

Comment:
Is this a requirement for translations? Why do we build the server?

Reply:
#7665 @seear 

It looks like this was an SSR thing, but with canaries we might just drop it?

I thought it might be related to i18n, but after looking at history I'm doubting that.

Reply:
It was to ensure that the SSR pragma check was run over the codebase during Circle CI tests. That checking was removed in #9585, so we will likely no longer need to build the server here.

Reply:
Great! Ripped out in 039b15c